### PR TITLE
PLT-1377: centralize & correct vSphere privileges

### PR DESCRIPTION
**Pull request by TylerGillson**
## Describe the Change

- De-quadruplicate vSphere privilege tables
- Alphabetize all tables & correct wording to match vSphere docs exactly (with a few minor casing exceptions)
- Add API-formatted privileges

Related:
- https://github.com/spectrocloud/palette-cli/pull/725
- https://github.com/validator-labs/validatorctl/pull/212

## Changed Pages

💻 [Add Preview URL for Page]()

## Jira Tickets

🎫 [Jira Ticket]()

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [ ] No. _Please leave a short comment below about why this PR cannot be backported._

### CODE REVIEW

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 1]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
 tags: ["data center", "vmware", "permissions"]
 ---
 
-The VMware vSphere user account that deploys host clusters require access to the following vSphere objects and
-permissions listed in the following table. Review the vSphere objects and privileges required to ensure each role is
-assigned the required privileges.
+The VMware vSphere user account that deploys host clusters and/or private cloud gateways requires all vSphere privileges
+listed in the following table on specific objects. A _Spectro root role_ must be created that contains each privilege in
+the table.
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:12]:
```suggestion
The VMware vSphere user account that deploys host clusters and/or private cloud gateways requires all vSphere privileges listed in the following section on specific objects.
```

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 2]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
...
 
 ### Spectro Root Role Privileges
 
-The spectro root role privileges are only applied to root objects and data center objects. Select the tab for the
-vSphere version you are using to view the required privileges for the spectro root role.
+Select the tab for the vSphere version you are using to view the required privileges for the _Spectro root role_.
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:16]:
```suggestion
A _Spectro root role_ must be created that contains each privilege in the following tables. Select the tab for the vSphere version you are using to view the required privileges.
```

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 3]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
...
+all required privileges on all required objects.
 
-:::
+| **vSphere Object**     | **Propagation** | **Role**            | **Condition**                                                            |
+| ---------------------- | --------------- | ------------------- | ------------------------------------------------------------------------ |
+| **vCenter Root**       | No              | _Spectro root role_ |                                                                          |
+| **Target Datacenter**  | No              | _Spectro root role_ |                                                                          |
+| **Target Cluster**     | No              | _Spectro root role_ |                                                                          |
+| **Target VM Folder**   | Yes             | _Spectro root role_ |                                                                          |
+| **Distributed Switch** | No              | _Spectro root role_ | If the Target Network is a Distributed Port Group residing beneath a VDS |
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:100]:
```suggestion
| **Distributed Switch** | No              | _Spectro root role_ | If the Target Network is a Distributed Port Group |
```

Distributed Port Group means it's necessarily a distributed switch.

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 4]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
...
 ### Spectro Role Privileges
 
-As listed in the table, apply spectro role privileges to vSphere objects you intend to use for Palette installation. A
-separate table lists Spectro role privileges for VMs by category.
-
-Open Virtual Appliance (OVA) files are downloaded to the folder you selected. These images are cloned from the folder
-and applied VMs that deployed during deployments.
+The VMware vSphere user account that deploys host clusters and/or private cloud gateways requires all vSphere privileges
+listed in the following table on specific objects. A _Spectro role_ must be created that contains each privilege in the
+following tables. A separate table lists Spectro role privileges for VMs by category.
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:106]:
```suggestion
A _Spectro role_ must be created that contains each privilege in the following tables. A separate table lists Spectro role privileges for VMs by category.
```

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 5]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
...
+| **Folder**                | Create folder<br />Delete folder<br />Move folder<br />Rename folder                                                                                                  |
+| **Host Local Operations** | Reconfigure virtual machine                                                                                                                                           |
+| **Network**               | Assign network                                                                                                                                                        |
+| **Resource**              | Apply recommendation<br />Assign virtual machine to resource pool<br />Migrate powered off virtual machine<br />Migrate powered on virtual machine<br />Query vMotion |
+| **Sessions**              | Validate session                                                                                                                                                      |
+| **spectro-templates**     | Read only. This is the vSphere folder created during the install. For airgap installs, you must manually create this folder.                                          |
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:123]: I think this is a typo as this isn't a permission but an entity on which to apply permission

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 6]
```diff
@@ -7,186 +7,231 @@ sidebar_position: 60
...
+| vSphere Replication   | Configure replication<br />Manage replication<br />Monitor replication                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:143]: I don't understand what this table is compared to the previous one?

**Review by cloudmaniac — COMMENTED** [review]
Looks better! I provided a few remarks and suggestions. Also, I think we're missing an introduction on why we have 2 roles (and I don't have this answer).

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 6 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md:143]: The _Spectro role_ tables are actually all identical. We just have three of them to support the tabbed "version selection".

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 5 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md]: Resolved in eff3548d1103ad25895211ffc3fc1076aabc3de1

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 4 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md]: Resolved in eff3548d1103ad25895211ffc3fc1076aabc3de1

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 3 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md]: Resolved in eff3548d1103ad25895211ffc3fc1076aabc3de1

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 2 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md]: Resolved in eff3548d1103ad25895211ffc3fc1076aabc3de1

File: docs/docs-content/clusters/data-center/vmware/permissions.md (hunk 1 above)
Comment (TylerGillson) [reply docs/docs-content/clusters/data-center/vmware/permissions.md]: Resolved in eff3548d1103ad25895211ffc3fc1076aabc3de1

File: docs/docs-content/clusters/data-center/vmware/permissions.md [hunk 7]
```diff
@@ -7,186 +7,227 @@ sidebar_position: 60
...
+| -------------------------------------------- | --------------- | -------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| **Target Network**                           | Yes             | _Spectro role_ |                                                                                                                                           |
+| **Target Datastore**                         | Yes             | _Spectro role_ |                                                                                                                                           |
+| **All ESXi hosts within the Target Cluster** | No              | _Spectro role_ |                                                                                                                                           |
+| **spectro-templates Folder**                 | Yes             | _Spectro role_ | For airgap installs, it must be manually created in advance and permissions assigned. For connected installs it is created automatically. |
```
Comment (TylerGillson) [inline docs/docs-content/clusters/data-center/vmware/permissions.md:232]: @karl-cardenas-coding I'd like to confirm this. I'm not aware of any automated mechanism by which this folder is created. I kinda think it's a manual process in all scenarios. But I'm probably wrong 🤷🏼‍♂️

Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/data-center/vmware/permissions.md:232]: You're right @TylerGillson

**Review by karl-cardenas-coding — APPROVED** [review]

**Pull request merged**